forward/forward_test.go

```diff
@@ -35,7 +35,8 @@ func TestCloneRequest(t *testing.T) {
 	const rawPath = "/%E4%B8%9C%E6%96%B9Project"
```

Comment:
By simply changing `rawPath` to `"/%E4%B8%9C%E6%96%B9Project/http%3A%2F%2Fwww.site.com%2Fsomething"`, the test fails in 1.4.3 but succeeds in 1.5. Other parts LGTM. @oxtoacart I'll go ahead to make that simple test change and merge.